pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+	vulnSummaryByNamespace.Namespace = namespace
+	vulnSummaryByNamespace.CreationTimestamp.Time = time.Now()
+
+	return vulnSummaryByNamespace
+}
+
+func updateVulnCounters(aggregatedCounters *softwarecomposition.VulnerabilityCounters, counters *softwarecomposition.VulnerabilityCounters) {
+	aggregatedCounters.All += counters.All
+	aggregatedCounters.Relevant += counters.Relevant
+}
+
+func updateSeverities(aggregatedSeverities *softwarecomposition.SeveritySummary, severities *softwarecomposition.SeveritySummary) {
+	updateVulnCounters(&aggregatedSeverities.Critical, &severities.Critical)
+	updateVulnCounters(&aggregatedSeverities.High, &severities.High)
+	updateVulnCounters(&aggregatedSeverities.Medium, &severities.Medium)
+	updateVulnCounters(&aggregatedSeverities.Low, &severities.Low)
+	updateVulnCounters(&aggregatedSeverities.Negligible, &severities.Negligible)
+	updateVulnCounters(&aggregatedSeverities.Unknown, &severities.Unknown)
+}
+
+func updateFullVulnSumm(fullVulnSumm *softwarecomposition.VulnerabilitySummary, vulnManifestSumm *softwarecomposition.VulnerabilityManifestSummary) {
+	updateSeverities(&fullVulnSumm.Spec.Severities, &vulnManifestSumm.Spec.Severities)
+	if vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj)
+	}
+	if vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj)
+	}
+}
+
+func (s *VulnSummaryStorageImpl) getVulnManifestSummaryDirPath(key string) string {
+	return filepath.Join(s.root, key, "..", "..", "..", "vulnerabilitymanifestsummaries")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) aggregateVulnSummaryOverNamespace(ctx context.Context, fullVulnSumm *softwarecomposition.VulnerabilitySummary, path string) error {
+	var err error
+
+	vulnSumm, err := s.getOneVulnManifestSummary(ctx, path)
+	if err != nil {
+		return err
+	}
+	updateFullVulnSumm(fullVulnSumm, vulnSumm)
+
+	return nil
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) aggregateVulnSummary(ctx context.Context, key string) (*softwarecomposition.VulnerabilitySummary, []error) {
+	var errs []error
+	fullVulnSumm := initNamespaceVulnSummary(key)
+
+	namespace := filepath.Base(key)
+	afero.Walk(s.appFs, s.getVulnManifestSummaryDirPath(key), func(path string, info os.FileInfo, err error) error {
+		if namespace != filepath.Base(filepath.Dir(path)) {
+			return nil
+		}
+		if !strings.HasSuffix(path, jsonExt) {
+			return nil
+		}
+		e := s.aggregateVulnSummaryOverNamespace(ctx, &fullVulnSumm, path)
+		if err != nil {
+			errs = append(errs, e)
+		}
+		return nil
+	})
```

Comment:
issue: this code deals with low-level details of how vulnerability manifests get stored

This is what I was afraid of: this Vulnerability Storage object now dives into the details of how Vulnerability Manifests are stored.

Consider this: if we change the `file.StorageImpl` to change the way it names the files it stores, we would have to change this function. This breaks at least the single responsibility principle, will mess up our code base and slow us down.

This function should have no idea how vulnerability manifests are stored. Instead, it should have a way to call the Vulnerability Manifest storage to fetch Vulnerability Manifests by the parameters it needs: namespace or entire cluster.  Then, it can be completely oblivious to how the Vulnerability Manifest storage is implemented, tolerate any future changes and still do its job efficiently.

One way of implementing this would be to add Vulnerability Manifest storage as a parameter to the Vulnerability Summary storage. Then, the Vulnerability Summary storage can call the Vulnerability Manifest storage to retrieve Vulnerability manifests. In pseudo-code, this idea would look like:
```go
vmStore := file.StorageImpl()
kind := "VulnerabilityManifest"
vms := vmStore.GetByNamespace(kind, namespace)
summary := SummarizeVulnerabilities(vms)
return summary
```